pkg/awsManager/ec2Manager.go

```diff
@@ -0,0 +1,122 @@
+package awsManager
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	clientpkg "github.com/openshift/aws-account-shredder/pkg/aws"
+	"os"
+	"strings"
+)
+
+// function to list all the EC2 instances
+// not removing this function right now , as it is easy to verify after deletion if it was actually deleted or not
+// keeping for future testing purpose ( easy to use )
+func ListEc2Instances(assumedRoleClient clientpkg.Client) {
+
+	token := ""
+	for {
+		ec2Descriptions, err := assumedRoleClient.DescribeInstances(&ec2.DescribeInstancesInput{NextToken: aws.String(token)})
+		if err != nil {
+			fmt.Println("ERROR:", err)
+			os.Exit(1)
+		}
+		fmt.Println("EC2 instances", ec2Descriptions)
+		if ec2Descriptions.NextToken != nil {
+			token = *ec2Descriptions.NextToken
+		} else {
+			break
+		}
+	}
+
+}
+
+// this lists all the instances that are eligible for deletion based on the tags and stored them in instances to be deleted
+// this only creates an array of pointers and does not delete the instances
+func ListEc2InstancesForDeletion(assumedRoleClient clientpkg.Client) []*string {
+
+	var EC2InstancesToBeDeleted []*string
+	token := ""
+	for {
+		ec2Descriptions, err := assumedRoleClient.DescribeInstances(&ec2.DescribeInstancesInput{NextToken: aws.String(token)})
+		if err != nil {
+			fmt.Println("ERROR:", err)
+			os.Exit(1)
+		}
+
+		// nested for loop to read the tags , as it is a part of structure inside a structure output. Refer : https://pkg.go.dev/github.com/aws/aws-sdk-go/service/ec2?tab=doc#DescribeInstancesOutput
+		for _, iterator := range ec2Descriptions.Reservations {
```

Comment:
`for _, reservation := ...`

Reply:
i have removed the func ListEc2Instances()